src/flareContract.ts

```diff
@@ -0,0 +1,169 @@
+import { getAddressBinderABI, getFlareContractRegistryABI, defaultContractAddresses, addressBinderContractName, validatorRewardManagerContractName } from "./flareContractConstants";
+import { prefix0x, saveUnsignedTxJson } from "./utils";
+
+/**
+ * @description checks if the address is registered with the addressBinder contract
+ * @param ethAddressToCheck
+ * @param network
+ */
+export async function isAddressRegistered(ethAddressToCheck: string, network: string): Promise<boolean> {
+
+  const rpcUrl = getRpcUrl(network)
+  const addressBinderContractAddress = await getContractAddress(network, addressBinderContractName)
+  const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+
+  const abi = getAddressBinderABI() as ethers.ContractInterface
+  const contract = new ethers.Contract(addressBinderContractAddress, abi, provider);
+
+  const result = await contract.cAddressToPAddress(ethAddressToCheck);
+
+  if (result !== '0x0000000000000000000000000000000000000000') {
+    console.log(`${colorCodes.greenColor}Address associated with key ${ethAddressToCheck}: ${result}${colorCodes.resetColor}`);
+    return true;
+  } else {
+    console.log(`${colorCodes.redColor}No address found for key ${ethAddressToCheck}${colorCodes.resetColor}`);
+    return false;
+  }
+}
+
+/**
+ * @description cregisters address  with the addressBinder contract
+ * @param {RegisterAddressInterface} addressParams
+ */
+export async function registerAddress(addressParams: RegisterAddressInterface) {
+  const { publicKey, pAddress, cAddress, network, wallet, derivationPath, pvtKey, transactionId } = addressParams
+
+  const rpcUrl = getRpcUrl(network)
+  const addressBinderContractAddress = await getContractAddress(network, addressBinderContractName)
+  const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+
+  const abi = getAddressBinderABI() as ethers.ContractInterface
+  const contract = new ethers.Contract(addressBinderContractAddress, abi, provider);
+
+  const checksumAddress = ethers.utils.getAddress(cAddress);
+  const nonce = await provider.getTransactionCount(checksumAddress);
+  const config: NetworkConfig = getConfig(network)
+
+  const pAddr = prefix0x(Buffer.from(bech32.fromWords(bech32.decode(pAddress.slice(2)).words)).toString('hex'));
+  const publicKeyPrefixed = prefix0x(publicKey)
+  const gasEstimate = await contract.estimateGas.registerAddresses(publicKeyPrefixed, pAddr, cAddress);
+  const gasPrice = await provider.getGasPrice();
+
+  const populatedTx = await contract.populateTransaction.registerAddresses(publicKeyPrefixed, pAddr, cAddress);
+  const unsignedTx = {
+    ...populatedTx,
+    nonce,
+    chainId: config.networkID,
+    gasPrice,
+    gasLimit: gasEstimate
+  }
+
+  const serializedUnsignedTx = ethers.utils.serializeTransaction(unsignedTx);
+  const txHash = ethers.utils.keccak256(serializedUnsignedTx);
+  const unsignedTxObj = createUnsignedJsonObject(txHash)
+
+  if (wallet == "Ledger") {
```

Comment:
you already have defined the Ledger , ForDefi, and private key as constant right. Please use those.